spec/integration/app_console_spec.rb

```diff
@@ -0,0 +1,60 @@
+RSpec.describe "app-(db)console" do
+  let(:cli) { GovukConnect::CLI.new }
+
+  before do
+    disable_any_exec(cli)
+    disable_yaml_load_file
+  end
+
+  it "SSHs and then runs govuk_app_console" do
+    node_classes = { "backend" => { "apps" => %w(my-app) } }
+
+    stub_puppet_hieradata(environment: :integration,
+                          node_classes: node_classes)
+
+    stub_govuk_node_list(machine_class: "backend",
+                         hostnames: %w(foo),
+                         environment: :integration)
+
+    args = ssh_command(environment: :integration,
+                       hostname: "foo",
+                       suffix: ["-t", "govuk_app_console my-app"])
+
+    allow(cli).to receive(:exec).with(*args)
```

Comment:
Could this be `expect(cli).to receive(:exec).with(*args)` to avoid needing line 25? similar for a few other specs.

Reply:
Unfortunately not. I had a hard time arriving at this pattern: we want to disable calls to `exec` in general (to avoid real SSH calls when running buggy tests), but then allow specific ones and `expect` that they happened. The problem is that `expect` doesn't enable the call to `exec`, so we need to do an explicit `allow` and then an `expect`.

(This is basically the same approach we take with WebMock.)